tests/apitests/python/test_retag.py

```diff
@@ -0,0 +1,128 @@
+        dst_repo_sub_name = "repo"
+        dst_tag_name = "test_tag"
+
+        #1. Create a new user(UA);
+        TestProjects.user_retag_id, user_retag_name = self.user.create_user(user_password = user_retag_password, **ADMIN_CLIENT)
+
+        TestProjects.USER_RETAG_CLIENT=dict(endpoint = url, username = user_retag_name, password = user_retag_password)
+
+        #2. Create a new project(PA) by user(UA);
+        TestProjects.project_src_repo_id, project_src_repo_name = self.project.create_project(metadata = {"public": "false"}, **TestProjects.USER_RETAG_CLIENT)
+
+        #3. Create a new project(PB) by user(UA);
+        TestProjects.project_dst_repo_id, project_dst_repo_name = self.project.create_project(metadata = {"public": "false"}, **TestProjects.USER_RETAG_CLIENT)
+
+        retag_member_id = self.project.get_project_member_id(TestProjects.project_dst_repo_id, user_retag_name, **TestProjects.USER_RETAG_CLIENT)
+
+        #4. Update role of user-retag as guest member of project(PB);
+        self.project.update_project_member_role(TestProjects.project_dst_repo_id, retag_member_id, 3, **ADMIN_CLIENT)
+
+        #5. Create a new repository(RA) in project(PA) by user(UA);
+        TestProjects.src_repo_name, tag_name = push_image_to_project(project_src_repo_name, harbor_server, 'admin', 'Harbor12345', "hello-world", pull_tag_name)
+
+        #6. Get repository in project(PA), there should be one repository which was created by user(UA);
+        src_repo_data = self.repo.get_repository(TestProjects.project_src_repo_id, **TestProjects.USER_RETAG_CLIENT)
+        _assert_status_code(TestProjects.src_repo_name, src_repo_data[0].name)
+
+        #7. Get repository(RA)'s image tag detail information;
+        src_tag_data = self.repo.get_tag(TestProjects.src_repo_name, tag_name, **TestProjects.USER_RETAG_CLIENT)
+
+        TestProjects.dst_repo_name = project_dst_repo_name+"/"+ dst_repo_sub_name
+
+        #8. Retag image in project(PA) to project(PB), it should be forbidden;
+        self.repo.retag_image(TestProjects.dst_repo_name, dst_tag_name, TestProjects.src_repo_name+":"+src_tag_data.digest, expect_status_code=403, **TestProjects.USER_RETAG_CLIENT)
+
+        #9. Update role of user-retag as admin member of project(PB);
+        self.project.update_project_member_role(TestProjects.project_dst_repo_id, retag_member_id, 1, **ADMIN_CLIENT)
+
+        #10. Retag image in project(PA) to project(PB), it should be successful;
+        self.repo.retag_image(TestProjects.dst_repo_name, dst_tag_name, TestProjects.src_repo_name+":"+src_tag_data.digest, **TestProjects.USER_RETAG_CLIENT)
+
+        #11. Get repository(RB)'s image tag detail information;
+        dst_tag_data = self.repo.get_tag(TestProjects.dst_repo_name, dst_tag_name, **TestProjects.USER_RETAG_CLIENT)
+
+        #12. Read digest of retaged image, it must be the same with the image in repository(RA);
+        _assert_status_code(src_tag_data.digest, dst_tag_data.digest)
```

Comment:
_assert_status_code == the name is confusing, acturally it's not for code assertion

Reply:
can we leverage the assertion in python?